MoEPluginBase/MoEMessageQueueHandler.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  MoEMessageQueueHandler.swift
+//  MoEPluginBase
+//
+//  Created by Rakshitha on 24/06/22.
+//
+
+import Foundation
+import MoEngageSDK
+
+@objc public protocol MoEPluginBridgeDelegate {
+    @objc func sendMessage(event: String, message: [String: Any])
+}
+
+final class MoEMessageQueueHandler {
+    private var identifier: String
+    private var messageQueue = [[String: Any]]()
+    private var isSDKInitialized = false
+    private weak var delegate: MoEPluginBridgeDelegate?
+    
+    init(identifier: String) {
+        self.identifier =  identifier
+    }
+    
+    func setBridgeDelegate(delegate: MoEPluginBridgeDelegate) {
+        self.delegate = delegate
+    }
+    
+    func flushMessage(eventName: String, message: [String: Any]? = nil) {
+        guard let message = message, !message.isEmpty else {
+            return
+        }
+
+        if isSDKInitialized {
+            delegate?.sendMessage(event: eventName, message: message)
+        } else {
+            let payload = [MoEPluginConstants.General.event: eventName, MoEPluginConstants.General.message: message] as [String: Any]
+            messageQueue.append(payload)
+        }
+    }
+    
+    func flushAllMessages() {
+        isSDKInitialized = true
+        
+        for messageQueue in messageQueue {
```

Comment:
@RakshithaAcharya can we name the array of dictionary as plural?

Reply:
@deepa-iOS-SDK  since `messagequeue` by its name reflects that it can hold multiple data(plural). However in the for loop I have replaced
 for payload in messageQueue { }
let me know your thoughts